csharp/src/Google.Protobuf/Reflection/Descriptor.cs

```diff
@@ -705,10 +705,10 @@ internal sealed partial class FileDescriptorProto : pb::IMessage<FileDescriptorP
       dependency_.Add(other.dependency_);
```

Comment:
Although there's no behavior difference, I sill prefer to change repeated_scalar_fields and repeated_enum_fields to call MergeFrom() as well

Reply:
A normal `Add` would probably be faster with scalar fields (since it'd just be an array copy) however that's an optimization that can be built into `MergeFrom`.

Reply:
I've fixed it. Only message type && not wrapper type use MergeFrom